src/components/bulk-email-tool/bulk-email-task-manager/test/BulkEmailContentHistory.test.jsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * @jest-environment jsdom
+ */
+import React from 'react';
+import {
+  render, screen, fireEvent, cleanup, act,
+} from '../../../../setupTest';
+import BulkEmailContentHistory from '../BulkEmailContentHistory';
+import { getSentEmailHistory } from '../data/api';
+import buildEmailContentHistoryData from '../data/__factories__/emailContentHistory.factory';
+
+jest.mock('../data/api', () => ({
+  __esModule: true,
+  getSentEmailHistory: jest.fn(() => {}),
+}));
+
+describe('BulkEmailContentHistory component', () => {
+  beforeEach(() => jest.resetModules());
+  afterEach(cleanup);
+
+  test('renders correctly', async () => {
+    render(<BulkEmailContentHistory />);
+    const tableDescription = await screen.findByText(
+      'To see the content of previously sent emails, click this button:',
+    );
+    expect(tableDescription).toBeTruthy();
+    const showEmailContentHistoryButton = await screen.findByText('Show Sent Email History');
+    expect(showEmailContentHistoryButton).toBeTruthy();
+  });
+
+  test('renders a table when the button is pressed and data is returned', async () => {
+    await act(async () => {
+      const emailHistoryData = buildEmailContentHistoryData(1);
+      getSentEmailHistory.mockImplementation(() => emailHistoryData);
+
+      render(<BulkEmailContentHistory />);
+
+      const showEmailContentHistoryButton = await screen.findByText('Show Sent Email History');
+      fireEvent.click(showEmailContentHistoryButton);
+
+      // verify component structure
+      const tableDescription = await screen.findByText(
+        'To read a sent email message, click the `View Message` button within the table.',
+      );
+      expect(tableDescription).toBeTruthy();
+
+      // verify table structure
+      expect(await screen.findByText('Subject')).toBeTruthy();
+      expect(await screen.findByText('Sent By')).toBeTruthy();
+      expect(await screen.findByText('Sent To')).toBeTruthy();
+      expect(await screen.findByText('Time Sent')).toBeTruthy();
+      expect(await screen.findByText('Number Sent')).toBeTruthy();
+
+      // verify table contents
+      const { emails } = emailHistoryData;
+      const email = emails[0];
+      expect(await screen.findByText(email.created)).toBeTruthy();
+      expect(await screen.findByText(email.number_sent)).toBeTruthy();
+      expect(await screen.findByText(email.requester)).toBeTruthy();
+      expect(await screen.findByText(email.sent_to.join(', '))).toBeTruthy();
+      expect(await screen.findByText(email.email.subject)).toBeTruthy();
+      expect(await screen.findAllByText('View Message')).toBeTruthy();
+
+      // verify modal behavior
+      const viewMessageButton = await screen.findByText('View Message');
```

Comment:
we may want to extract this portion of logic out onto its own test, or change the description of this test to reflect the modal interaction. Overall I'd consider the modal its own "unit".

Reply:
Fair enough!

Reply:
I moved pieces that verified the modal behavior and content into its own unit test. Thanks for the feedback! :)